### ISSUE TITLE
Fix Request: AvoidUnguardedMutableInheritedFieldsInSharedObjects false positive in for/switch/... blocks

### ISSUE THREAD

**Issue by stokpop** [jborgers/PMD-jPinpoint-rules#502]
```java
@Component
public final class Foo extends AbstractService {
    public void importResources(List<Resource> resources, User user) throws Exception {
        for (ResourceData resourceData : resources) {
            resourceData.setNewLink(new Link(user, resourceData)); // <- false positive
        }
    }
    public void handle(Msg event) {
        Id id = event.getId();
        switch (id.getEntityType()) { // switch is needed to get false positive
            case TENANT:
                Set<Notification> toCancel = new HashSet<>();
                requests.forEach((requestId, metadata) -> {
                    if (metadata.getId().equals(id)) {
                        toCancel.add(requestId); // <- false positive
                    }
                });
                break;
        }
    }
}
```
